src/components/EventDetails.spec.tsx

```diff
@@ -0,0 +1,137 @@
+            });
+        cy.fixture('dummyEventResponses.json')
+            .then((data : {data: ZetkinEventResponse[]}) => {
+                dummyEventResponse = data.data[0];
+            });
+    });
+
+    it('contains data for each event', () => {
+        mountWithProviders(
+            <EventDetails
+                event={ dummyEvent }
+                onEventResponse={ () => null }
+                org={ dummyOrg }
+                response={ dummyEventResponse }
+            />,
+        );
+        cy.get('[data-test="event-title"]').should('be.visible');
+        cy.get('[data-test="org-title"]').should('be.visible');
+        cy.get('[data-test="campaign-title"]').should('be.visible');
+        cy.get('[data-test="event-dates"]').should('be.visible');
+        cy.get('[data-test="event-times"]').should('be.visible');
+        cy.get('[data-test="location"]').should('be.visible');
+        cy.get('[data-test="info-text"]').should('be.visible');
+    });
+
+    it('does not show event info if none is provided', () => {
+        dummyEvent.info_text = '';
+        mountWithProviders(
+            <EventDetails
+                event={ dummyEvent }
+                onEventResponse={ () => null }
+                org={ dummyOrg }
+                response={ dummyEventResponse }
+            />,
+        );
+        cy.get('[data-test="info-text"]').should('not.be.visible');
+    });
+
+    it('contains an activity title instead of missing event title', () => {
+        dummyEvent.title = undefined;
+        mountWithProviders(
+            <EventDetails
+                event={ dummyEvent }
+                onEventResponse={ () => null }
+                org={ dummyOrg }
+                response={ dummyEventResponse }
+            />,
+        );
+        cy.get('[data-test="event-title"]')
+            .should('contain', dummyEvent.activity.title)
+            .should('not.contain', 'undefined');
+    });
+
+    it('contains a link back to the organisation page', () => {
+        mountWithProviders(
+            <EventDetails
+                event={ dummyEvent }
+                onEventResponse={ () => null }
+                org={ dummyOrg }
+                response={ dummyEventResponse }
+            />,
+        );
+        cy.get('[data-test="org-title"]')
+            .should('have.attr', 'href', `/o/${dummyOrg.id}`);
+    });
+
+    it('contains a link back to the campaign page', () => {
+        mountWithProviders(
+            <EventDetails
+                event={ dummyEvent }
+                onEventResponse={ () => null }
+                org={ dummyOrg }
+                response={ dummyEventResponse }
+            />,
+        );
+        cy.get('[data-test="campaign-title"]')
+            .should('have.attr', 'href', `/o/${dummyOrg.id}/campaigns/${dummyEvent.campaign.id}`);
+    });
+
+    it('contains a sign-up button for the event', () => {
+        const spyOnSubmit = cy.spy();
```

Comment:
I think this should be named `spyOnEventResponse` (or `onEventResponseSpy`) instead.

Reply:
Same goes for the spy in the other tests, of course.